paasta_tools/contrib/rightsizer_soaconfigs_update.py

```diff
@@ -183,33 +188,35 @@ class KubernetesRightsizerResult(TypedDict):
     suggested_max_instances: int
 
 
-class KubernetesRecommendation(TypedDict):
-    disk: str
-    mem: str
+class CPU(TypedDict):
+    cpu: float
+
+
+class Hacheck(TypedDict):
+    requests: CPU
+    limits: CPU
+
+
+class SidecarResourceRequirements(TypedDict):
+    hacheck: Hacheck
+
+
+class KubernetesRecommendation(TypedDict, total=False):
+    disk: float
+    mem: float
     cpus: float
     cpu_burst_add: float
     max_instances: int
     min_instances: int
-    sidecar_resource_requirements: TypedDict(
-        "SidecarResourceRequirements",
-        {
-            "hacheck": TypedDict(
-                "Hacheck",
-                {
-                    "requests": TypedDict("Requests", {"cpu": float}),
-                    "limits": TypedDict("Limits", {"cpu": float}),
-                },
-            ),
-        },
-    )
+    sidecar_resource_requirements: SidecarResourceRequirements
```

Comment:
oh, looking closer at what we do in `kubernetes_tools.py`:
```suggestion
    sidecar_resource_requirements: Dict[str, SidecarResourceRequirements]
```
should work - this lets us get rid of the 3 typeddicts above

(note: this will require `from paasta_tools.kubernetes_tools import SidecarResourceRequirements`)